#! /usr/bin/env python

# 1.导包
from dynamic_reconfigure.parameter_generator_catkin import *
PACKAGE = "hero_chassis_controller"
# 2.创建生成器
gen = ParameterGenerator()

# 3.向生成器添加若干参数
#add(name, paramtype, level, description, default=None, min=None, max=None, edit_method="")
gen.add("front_left_p", double_t, 0, "P gain for left front wheel", 0.4, -10, 10)
gen.add("front_left_i", double_t, 0, "I gain for left front wheel", 0.1, -10, 10)
gen.add("front_left_d", double_t, 0, "D gain for left front wheel", 0, -10, 10)

gen.add("front_right_p", double_t, 0, "P gain for right front wheel", 0.4, -10, 10)
gen.add("front_right_i", double_t, 0, "I gain for right front wheel", 0.1, -10, 10)
gen.add("front_right_d", double_t, 0, "D gain for right front wheel", 0, -10, 10)

gen.add("back_left_p", double_t, 0, "P gain for left back wheel", 0.4, -10, 10)
gen.add("back_left_i", double_t, 0, "I gain for left back wheel", 0.1, -10, 10)
gen.add("back_left_d", double_t, 0, "D gain for left back wheel", 0, -10, 10)

gen.add("back_right_p", double_t, 0, "P gain for right back wheel", 0.4, -10, 10)
gen.add("back_right_i", double_t, 0, "I gain for right back wheel", 0.1, -10, 10)
gen.add("back_right_d", double_t, 0, "D gain for right back wheel", 0, -10, 10)

# 4.生成中间文件并退出
exit(gen.generate(PACKAGE,"pidNode","PID"))
